definitions/npm/react-redux_v5.x.x/flow_v0.89.x-/test_connect.js

```diff
@@ -0,0 +1,495 @@
+  //$ExpectError forMapDispatchToProps missing
+  <Connected passthrough={123} forMapStateToProps={'data'} />;
+}
+
+function testMapDispatchToPropsWithoutMapStateToProps() {
+  type Props = {
+    passthrough: number,
+    fromMapDispatchToProps: string
+  };
+  class Com extends React.Component<Props> {
+    render() {
+      return <div>
+        {this.props.passthrough}
+        {this.props.fromMapDispatchToProps}
+      </div>;
+    }
+  }
+
+  type MapDispatchToPropsProps = {forMapDispatchToProps: string};
+  const mapDispatchToProps = (dispatch: *, ownProps: MapDispatchToPropsProps) => {
+    return {fromMapDispatchToProps: ownProps.forMapDispatchToProps}
+  }
+  const Connected = connect(null, mapDispatchToProps)(Com);
+  <Connected passthrough={123} forMapStateToProps={'data'} forMapDispatchToProps={'more data'} />;
+  //$ExpectError passthrough missing
+  <Connected forMapStateToProps={'data'} forMapDispatchToProps={'more data'} />;
+  //$ExpectError forMapDispatchToProps missing
+  <Connected passthrough={123} forMapStateToProps={'data'} />;
+}
+
+function testMapDispatchToPropsPassesActionCreators() {
+  type Props = {
+    passthrough: number,
+    dispatch1: (num: number) => void,
+    dispatch2: () => void
+  };
+  class Com extends React.Component<Props> {
+    render() {
+      return <div>{this.props.passthrough}</div>;
+    }
+  }
+
+  const mapDispatchToProps = {
+    dispatch1: (num: number) => {},
+    dispatch2: () => {}
+  };
+  const Connected = connect(null, mapDispatchToProps)(Com);
+  <Connected passthrough={123}/>;
+  //$ExpectError no passthrough
+  <Connected/>;
+
+  const mapDispatchToPropsWithoutDispatch2 = {
+    dispatch1: (num: number) => {}
+  };
+  const Connected2 = connect(null, mapDispatchToPropsWithoutDispatch2)(Com);
+  //$ExpectError no dispatch2
+  <Connected2 passthrough={123}/>;
+
+  const mapDispatchToPropsWithWrongDispatch1 = {
+    //$ExpectError dispatch1 should be number
+    dispatch1: (num: string) => {},
+    dispatch2: () => {}
+  };
+  const Connected3 = connect(null, mapDispatchToPropsWithWrongDispatch1)(Com);
+  <Connected3 passthrough={123}/>;
+}
+
+function testMapDispatchToPropsPassesActionCreatorsWithMapStateToProps() {
+  type Props = {
+    passthrough: number,
+    dispatch1: () => void,
+    dispatch2: () => void,
+    fromMapStateToProps: number
+  };
+  class Com extends React.Component<Props> {
+    render() {
+      return <div>{this.props.passthrough}</div>;
+    }
+  }
+  type State = {a: number}
+  type MapStateToPropsProps = {forMapStateToProps: string}
+  const mapStateToProps = (state: State, props: MapStateToPropsProps) => {
+    return {
+      fromMapStateToProps: state.a
+    }
+  }
+  const mapDispatchToProps = {
+    dispatch1: () => {},
+    dispatch2: () => {}
+  };
+  const Connected = connect(mapStateToProps, mapDispatchToProps)(Com);
+  <Connected passthrough={123} forMapStateToProps="str"/>;
+  //$ExpectError no passthrough
```

Comment:
getting

```
    
    Cannot create `Connected` element because property `forMapStateToProps` is missing in props [1] but exists in
    `MapStateToPropsProps` [2] in the second argument.
```

instead of error about passtrough. Might be that `<Connected/>;` should have everything except `passtrough` defined here instead.